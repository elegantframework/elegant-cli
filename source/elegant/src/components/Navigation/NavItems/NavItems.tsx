import { NavigationItem } from "@/types/NavigationItem";
import clsx from "clsx";
import Link from "next/link";
import { useRouter } from "next/router";

interface Props {
    /**
     * A list of navigation items
     */
    navItems: NavigationItem[];
};

/**
 * A list of navigation items for the header navigation element.
 * @returns A list of navigation items.
 */
const NavItems = ({navItems}: Props) => {
    const router = useRouter();

    let list: JSX.Element[] = [];

    // loop through our items, and add them to our list
    navItems.forEach(item => {
        list.push(
            <li key={item.label}>
                <Link
                    href={item.href}
                    className={
                        clsx(
                            'hover:text-primary-500 dark:hover:text-primary-400', 
                            (router.pathname.indexOf(item.path) > -1 ? 'text-primary-500 dark:text-primary-400' : '')
                        )
                    }
                >
                    {item.label}
                </Link>
            </li>
        ); 
    });

    return(<>{list}</>);
};

export default NavItems;