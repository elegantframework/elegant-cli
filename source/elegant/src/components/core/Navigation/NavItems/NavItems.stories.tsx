import type { Meta, StoryObj } from '@storybook/react';
import NavItems from './NavItems';

const meta: Meta<typeof NavItems> = {
  title: 'Core/Navigation/Nav Items',
  component: NavItems
};

export default meta;

type Story = StoryObj<typeof NavItems>;

export const Default: Story = {
  args: {
    navItems: [
        {
            href: "/docs/installation",
            path: "/docs/",
            label: "Docs",        
        },
        {
            href: "/blog",
            path: "/blog",
            label: "Blog",        
        }
    ]
  },
};