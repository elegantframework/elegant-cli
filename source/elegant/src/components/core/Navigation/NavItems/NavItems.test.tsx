import React from "react";
import renderer from 'react-test-renderer';
import NavItems from "./NavItems";

describe("Nav Item component", () => {

    const useRouter = jest.spyOn(require("next/router"), "useRouter");

    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn()
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null)
    }));

    it('renders the nav items component properly', () => {
        const navItem = renderer
        .create(
            <NavItems 
                navItems={[
                    {
                        href: "/docs/installation",
                        path: "/docs/",
                        label: "Docs",        
                    },
                    {
                        href: "/blog",
                        path: "/blog",
                        label: "Blog",        
                    }
                ]}
            />
        )
        .toJSON();
        expect(navItem).toMatchSnapshot();
    });

    it('renders the nav items component properly with no nav items', () => {
        const navItem = renderer
        .create(
            <NavItems 
                navItems={[]}
            />
        )
        .toJSON();
        expect(navItem).toMatchSnapshot();
    });
});