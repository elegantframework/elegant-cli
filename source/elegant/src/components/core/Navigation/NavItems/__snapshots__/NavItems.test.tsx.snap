// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Item component renders the nav items component properly 1`] = `
[
  <li>
    <a
      className="hover:text-primary-500 dark:hover:text-primary-400"
      href="/docs/installation"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Docs
    </a>
  </li>,
  <li>
    <a
      className="hover:text-primary-500 dark:hover:text-primary-400"
      href="/blog"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Blog
    </a>
  </li>,
]
`;

exports[`Nav Item component renders the nav items component properly with no nav items 1`] = `null`;
